### PR TITLE
[chore] rewritting depenencies with `uv`, add configuration for reranker benchmark

### PR DESCRIPTION
**Pull request by sigridjineth**
## Changes
1. Added `uv` as a project management tool to provide better organization and long-term maintainability.
2. Implemented configurable options for evaluating rerankers, maintaining the existing OpenAI embedding model support and simultaneously enabling comprehensive cross-encoder configurations with hf's `trust_remote_code` toggle.

### CODE REVIEW

**Comment by sigridjineth** [comment]
cc. @vkehfdl1

File: .gitignore [hunk 1]
```diff
@@ -166,3 +166,5 @@ data/**/*.csv
 data/*.csv
 
 project_dir/
+
+.python_version
```
Comment (vkehfdl1) [inline .gitignore:170]: Please add new line

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[project]
+name = "autorag-example-korean-embedding-benchmark"
+version = "0.1.0"
+description = "Add your description here"
+readme = "README.md"
+requires-python = ">=3.11"
```
Comment (vkehfdl1) [inline pyproject.toml:6]: We need more than python 3.10

File: config/embedding_benchmark.yaml [hunk 3]
```diff
@@ -9,19 +9,4 @@ node_lines:
...
-           - paraphrase-multilingual-mpnet-base-v2 # sentence-transformers/paraphrase-multilingual-mpnet-base-v2
-           - paraphrase-multilingual-MiniLM-L12-v2 # sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2
-           - multilingual-e5-large-instruct # intfloat/multilingual-e5-large-instruct
-           - KU-HIAI-ONTHEIT-large-v2 # KU-HIAI-ONTHEIT/ontheit-large-v2 (private)
-           - KU-HIAI-ONTHEIT-large-v2.1 # KU-HIAI-ONTHEIT/ontheit-large-v1_1 (private)
-           - kf-deberta-multitask # upskyy/kf-deberta-multitask
-           - gte-multilingual-base # Alibaba-NLP/gte-multilingual-base
-           - bge-m3 # BAAI/bge-m3
-           - KoE5 # nlpai-lab/KoE5
+           - gte-multilingual-reranker-base
```
Comment (vkehfdl1) [inline config/embedding_benchmark.yaml:12]: Do not change this since this is embedding benchmark repository

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]
Why don't we use uv at the AutoRAG main repo also?

File: config/embedding_benchmark.yaml (hunk 3 above)
Comment (realsigridjin) [reply config/embedding_benchmark.yaml:12]: @vkehfdl1 clearly mistake. gotcha

**Pull request closed without merging**